### ISSUE TITLE
libc: symlink creation fails with "deadlock ahead" message

### ISSUE THREAD

**Issue by nfeske** [genodelabs/genode#4219]
Thanks to Chrissimir for reporting an error during the creation of symlinks: The problem can be reproduced by issuing the following commands via the bash.run script:

```
$ cd /home
$ ln -s a b
```

The following message shows up in the log:

```
[init -> /bin/bash -> 7] Error: deadlock ahead, mutex=0x10ff8c70, return ip=0x500583a7
```

**Comment by nfeske** [comment]
Commit https://github.com/genodelabs/genode/commit/6a3a8ee4b17f73e94809854f95e244a66bb57b09 addresses this problem.

**Comment by nfeske** [comment]
Commit https://github.com/genodelabs/genode/commit/089beaa5b512aa983439b1476ee75493a9b46a44 adds a regression test to the tool_chain_auto.run scenario. Granted, it is not the ideal place for regression tests like this but after revisiting a few other candidates (e.g., test-libc_vfs), I found the problem easiest to trigger in the tool-chain test.

I also tried enhancing the extract.run scenario in this respect (adding an archive with a symlink) with the intention of triggering this issue. However, it turns out that the extract tool backs out from handling symlinks for other reasons. Just for keeping the record, the call of `archive_write_header`  at https://github.com/genodelabs/genode/blob/master/repos/libports/src/app/extract/main.cc#L197 returns ARCHIVE_FAILED when encountering the symlink. Given the can of worms that symlinks are (e.g., https://securitytracker.com/id/1036978), it is probably the safest to keep symlink support out of the extract tool anyway.
